File: frontend/src/api/salary.ts
import axios from "axios";

const API_BASE = import.meta.env.VITE_API_URL || "http://localhost:8000";

export interface SalaryRange {
  min: number;
  max: number;
  currency: string;
  period: string;
}

export interface SalaryResponse {
  role: string;
  location: string;
  experience_level: string;
  salary_range: SalaryRange;
  market_context: string;
  top_companies: string[];
  negotiation_tips: string[];
  confidence: "low" | "medium" | "high";
  disclaimer: string;
}

export interface SalaryRequest {
  role: string;
  location: string;
  experience_level: string;
  skills: string[];
}

export const estimateSalary = async (
  data: SalaryRequest,
  token: string
): Promise<SalaryResponse> => {
  const res = await axios.post(`${API_BASE}/salary/estimate`, data, {
    headers: { Authorization: `Bearer ${token}` },
  });
  return res.data;
};

File: backend/app/schemas/salary.py
from pydantic import BaseModel
from typing import List

class SalaryRequest(BaseModel):
    role: str
    location: str
    experience_level: str  # "entry", "mid", "senior", "lead"
    skills: List[str]

class SalaryRange(BaseModel):
    min: int
    max: int
    currency: str
    period: str  # "per year", "per month"

class SalaryResponse(BaseModel):
    role: str
    location: str
    experience_level: str
    salary_range: SalaryRange
    market_context: str
    top_companies: List[str]
    negotiation_tips: List[str]
    confidence: str  # "low", "medium", "high"
    disclaimer: str